trustd/src/main.rs

```diff
@@ -25,119 +23,52 @@ pub enum Command {}
 pub struct Trustd {
     #[command(subcommand)]
     pub(crate) command: Option<Command>,
-
-    /// The database creation mode
-    #[arg(long, env, value_enum, default_value_t = CreationMode::Default)]
-    pub creation: CreationMode,
-
-    #[arg(long, env, default_value_t = true, requires = "auth")]
-    pub with_http: bool,
-
-    #[arg(long, env)]
-    pub devmode: bool,
-
-    #[command(flatten)]
-    pub storage: StorageConfig,
-
-    #[command(flatten)]
-    pub database: Database,
-
-    #[command(flatten)]
-    pub infra: InfrastructureConfig,
-
-    #[command(flatten)]
-    pub auth: AuthConfigArguments,
-
-    #[command(flatten)]
-    pub http: HttpServerConfig<Trustify>,
-
-    #[command(flatten)]
-    pub swagger_ui_oidc: SwaggerUiOidcConfig,
 }
 
 impl Trustd {
     async fn run(self) -> ExitCode {
         match self.run_command().await {
             Ok(code) => code,
             Err(err) => {
-                log::error!("Error: {err}");
+                eprintln!("Error: {err}");
```

Comment:
Hah, I suggested `eprintln!(...)`

Reply:
I think this would be wrapped by the `Infrastructure` stuff. Or, alternatively, errors should bubble up to `main`, and there we could do the "catch all" error reporting.

Reply:
Or are you suggesting we *only* have the `Infrastructure` stuff in `trustd` and nowhere else? Because that would involve a lot of rework in `server/src/lib.rs`.

Reply:
Recounting here the results of an external chat:

- have the infrastructure in the actual (sub-)command
- the top level main would log terminal errors using eprintln
- other, more cli like commands, can use env_logger

In a nutshell:

- cli sub-commands use env_logger
- server style commands use infrastructure